moveit_ros/planning/rdf_loader/src/rdf_loader.cpp

```diff
@@ -62,68 +62,51 @@ RDFLoader::RDFLoader(const std::shared_ptr<rclcpp::Node>& node, const std::strin
   moveit::tools::Profiler::ScopedStart prof_start;
   moveit::tools::Profiler::ScopedBlock prof_block("RDFLoader(robot_description)");
 
-  auto start = std::chrono::system_clock::now();
+  auto start = node->now();
 
-  std::string content;
-  // TODO(JafarAbdi): Revise parameter lookup
-  if (node->has_parameter(robot_description))
-  {
-    rclcpp::Parameter robot_description_param = node->get_parameter(robot_description);
-    try
-    {
-      content = robot_description_param.as_string();
-    }
-    catch (const rclcpp::ParameterTypeException& e)
-    {
-      RCLCPP_WARN(LOGGER, "When getting robot_description parameter %s", e.what());
-    }
-  }
-  if (content.empty())
+  // Check if the robot_description parameter is declared, declare it if it's not declared yet
+  if (!node->has_parameter(robot_description))
+    node->declare_parameter(robot_description);
+  std::string robot_description_content;
+  node->get_parameter_or(robot_description, robot_description_content, std::string());
+
+  if (robot_description_content.empty())
   {
     RCLCPP_INFO_ONCE(LOGGER, "Robot model parameter not found! Did you remap '%s'?\n", robot_description.c_str());
     return;
   }
 
   urdf::Model* umodel = new urdf::Model();
-  if (!umodel->initString(content))
+  if (!umodel->initString(robot_description_content))
   {
-    RCLCPP_INFO(LOGGER, "Unable to parse URDF from parameter: '%s'", robot_description_.c_str());
+    RCLCPP_INFO(LOGGER, "Unable to parse URDF from parameter: '%s'", robot_description.c_str());
     return;
   }
   urdf_.reset(umodel);
 
-  const std::string srdf_description(robot_description_ + "_semantic");
-  std::string scontent;
-  // TODO(JafarAbdi): Revise parameter lookup
-  if (node->has_parameter(srdf_description))
-  {
-    rclcpp::Parameter srdf_description_param = node->get_parameter(srdf_description);
-    try
-    {
-      content = srdf_description_param.as_string();
-    }
-    catch (const rclcpp::ParameterTypeException& e)
-    {
-      RCLCPP_WARN(LOGGER, "When getting robot_description_semantic parameter: %s", e.what());
-    }
-  }
-  if (scontent.empty())
+  const std::string srdf_description = robot_description + "_semantic";
+  // Check if the robot_description_semantic parameter is declared, declare it if it's not declared yet
+  if (!node->has_parameter(srdf_description))
```

Comment:
In general, I don't think we should check `has_parameter()`, except we actually have shared parameters with other classes or instances. It would be much more verbose if we would call `declare_parameter()` with a good description while implicitly disallowing other classes to declare the same parameters. This would follow a single-responsibility principle. Also, `declare_parameter()` already returns the value we need and allows us to set the default, making `get_parameter_or()` redundant. We can keep it in this PR for now, but this is something I would like to refactor for the whole code base in order to really clean up and structure parameter use.

Reply:
The reason I add the checking for a previous declaration is that in Rviz When I add multiple RobotState plugins it was throwing an exception